nmigen/hdl/rec.py

```diff
@@ -113,6 +113,8 @@ def concat(a, b):
         return Record(other.layout, name=new_name, fields=fields, src_loc_at=1)
 
     def __init__(self, layout, *, name=None, fields=None, src_loc_at=0):
+        super().__init__(src_loc_at=src_loc_at)
```

Comment:
I think this needs a `+ 1`, but please check if the source locations are in fact correct.
```suggestion
        super().__init__(src_loc_at=src_loc_at + 1)
```

Reply:
will check

Reply:
it works without +1. with +1 i get ValueError: call stack is not deep enough. without the source location is correct

Reply:
Ah, it looks like I already added this in the `UserValue` constructor.